data_loader/8_fakenews/fakenews_plugin.py

```diff
@@ -0,0 +1,167 @@
+"""Load data from the Squirro Fake News Training API"""
+
+import logging
+import hashlib
+import json
+import requests
+import urllib.parse
+from squirro.dataloader.data_source import DataSource
+
+log = logging.getLogger(__name__)
+
+
+class FakeNewsDataSource(DataSource):
+    """FakeNews Plugin for the sq_data_loader.
+
+    Features:
+    * Load a single section of the news
+    * Full and Incremental load
+    * Web UI and CLI Support
+    """
+
+    def __init__(self):
+        pass
+
+    def connect(self, inc_column=None, max_inc_value=None):
+        if self.args.reset:
+            log.info("Resetting key-value stores")
+            self.key_value_cache.clear()
+            self.key_value_store.clear()
+
+        # create a requests session object so we get connection pooling
+        self.requests = requests.Session()
+
+    def disconnect(self):
+        # close the https sesssion
+        self.requests.close()
+
+    def getDataBatch(self, batch_size=100):
+
+        items = []
+
+        # iterate all requests sections
+        for section in self.args.section:
+            # continue from last run or if not present start from 0
+            since = self.key_value_store.get(section, 0)
```

Comment:
We should really implement incremental loading using the built-in methods, rather than recreating some mechanism every time.

Reply:
How do you suggest one tracks multiple values with the built in option?

Again, i'll do an example with the built in one, but i'm under the assumption that once a plugin in one job has to track multiple values or something more complex than a sortable id, string, date, then self._key_value_store is the way to go.